File: 알고리즘/2/Valid Palindrome.cpp
#include <string>

using namespace std;

class Solution {
public:
    char asciilower(char in){
        if('A'<=in&&in<='Z') return in - ('Z'-'z');
        else return in;
    }

    bool isPalindrome(string s) {
        int pointer1 = 0;
        int pointer2 = s.size()-1;
        char a =asciilower(s[pointer1]);
        char b=asciilower(s[pointer2]);
        while true{
            if(pointer1>=pointer2) return true;
            if(a==b){
                pointer1+=1;
                pointer2-=1;
                a = s[pointer1];
                while a==' '{
                    p
                }
                b = s[pointer2];
            }

        }
    }
};

File: 알고리즘/2/String/Valid Palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        list1 = [ele.lower() for ele in s if ele.isalnum()]

        left = 0
        right = len(list1) - 1
        while len(list1) > 1:
            if list1.pop(0) != list1.pop(-1):
                return False

        return True

    def isPalindrome2(self, s: str) -> bool:
        import re
        s = re.sub('[^a-z0-9]', '', s)

        return s[::] ==s[::-1]
        
                